### PR TITLE
dotCMS/core#17982 Update push publish dialog to use Angular one

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/api/services/dot-push-publish-dialog/dot-push-publish-dialog.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+@Injectable()
+export class DotPushPublishDialogService {
+    private _showDialog: Subject<PushPublishEvent> = new Subject<PushPublishEvent>();
+
+    get showDialog$(): Observable<PushPublishEvent> {
+        return this._showDialog.asObservable();
+    }
+
+    openDialog(data: PushPublishEvent): void {
```
Comment (fmontes) [inline src/app/api/services/dot-push-publish-dialog/dot-push-publish-dialog.service.ts]:
```suggestion
    open(data: PushPublishEvent): void {
```

We already have the context.

File: src/app/api/services/push-publish/push-publish.service.ts [hunk 2]
```diff
@@ -37,15 +38,15 @@ export class PushPublishService {
      */
     getEnvironments(): Observable<DotEnvironment[]> {
         return this.currentUser.getCurrentUser().pipe(
-            mergeMap((user) => {
+            mergeMap(user => {
```
Comment (fmontes) [inline src/app/api/services/push-publish/push-publish.service.ts]: Can we add type here?

File: src/app/api/services/push-publish/push-publish.service.ts [hunk 3]
```diff
@@ -37,15 +38,15 @@ export class PushPublishService {
...
                 return this.coreWebService.requestView({
                     method: RequestMethod.Get,
                     url: `${this.pushEnvironementsUrl}/${user.roleId}/name=0`
                 });
             }),
             pluck('bodyJsonObject'),
             mergeMap((environments: DotEnvironment[]) => environments),
-            filter((environment) => environment.name !== ''),
+            filter(environment => environment.name !== ''),
```
Comment (fmontes) [inline src/app/api/services/push-publish/push-publish.service.ts]: and type here?

File: src/app/portlets/shared/dot-content-types-listing/dot-content-types.component.ts [hunk 4]
```diff
@@ -80,22 +79,19 @@ export class DotContentTypesPortletComponent implements OnInit {
...
             this.dotLicenseService.isEnterprise(),
-            this.pushPublishService.getEnvironments().pipe(
-                map((environments: DotEnvironment[]) => !!environments.length),
-                take(1)
-            ),
-            this.route.data.pipe(
-                pluck('filterBy'),
-                take(1)
-            )
+            this.pushPublishService
```
Comment (fmontes) [inline src/app/portlets/shared/dot-content-types-listing/dot-content-types.component.ts]: How many columns do you have setup in your editor?, is weird that this is happening.

File: src/app/shared/models/push-publish-data/push-publish-data.ts [hunk 5]
```diff
@@ -13,3 +13,10 @@ export interface PushPublishData {
     forcePush: boolean;
     filterKey: string;
 }
+
+export interface PushPublishEvent {
```
Comment (fmontes) [inline src/app/shared/models/push-publish-data/push-publish-data.ts]:
```suggestion
export interface DotPushPublishEvent {
```

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts [hunk 6]
```diff
@@ -30,74 +32,40 @@ export class DotPushPublishContentTypesDialogComponent implements OnInit, OnDest
...
-            'contenttypes.content.push_publish.publish_date_errormsg',
-            'contenttypes.content.push_publish.expire_date_errormsg'
-        ]);
-        const filterOptions$ = this.dotPushPublishFiltersService
-            .get()
-            .pipe(catchError(() => of([])));
-
-        combineLatest(messages$, filterOptions$)
+        this.loadMessagesAndFilters();
+        this.dotPushPublishDialogService.showDialog$
```
Comment (fmontes) [inline src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: `takeUntil`?

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts [hunk 7]
```diff
@@ -111,6 +79,7 @@ export class DotPushPublishContentTypesDialogComponent implements OnInit, OnDest
      */
     close(): void {
         this.cancel.emit(true);
+        this.dialogShow = false;
```
Comment (fmontes) [inline src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: Shouldn't call the service here?

File: src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts [hunk 8]
```diff
@@ -21,15 +22,17 @@ export class DotIframeEventsHandler {
...
     ) {
         if (!this.handlers) {
             this.handlers = {
                 'edit-page': this.goToEditPage.bind(this),
                 'edit-contentlet': this.editContentlet.bind(this),
                 'edit-task': this.editTask.bind(this),
                 'create-contentlet': this.createContentlet.bind(this),
-                'company-info-updated': this.setDotcmsUiColors.bind(this)
+                'company-info-updated': this.setDotcmsUiColors.bind(this),
+                'push-publish-dialog': this.pushPublishDialog.bind(this)
```
Comment (fmontes) [inline src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts]:
```suggestion
                'push-publish': this.pushPublishDialog.bind(this)
```

Is the name of the event not the component.

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts (hunk 6 above)
Comment (hmoreras) [reply src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: is in the next line

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts (hunk 7 above)
Comment (hmoreras) [reply src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: the show hide of the dialog lives in the component.  the service the only send data.

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.spec.ts [hunk 9]
```diff
@@ -30,8 +30,8 @@ class TestHostComponent {
     pushPublishIdentifier: string;
 }
 
-xdescribe('DotPushPublishContentTypesDialogComponent', () => {
-    let comp: DotPushPublishContentTypesDialogComponent;
+xdescribe('DotPushPublishDialogComponent', () => {
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.spec.ts]: xdescribe?

File: projects/dotcms-js/src/lib/core/core-web.service.ts [hunk 10]
```diff
@@ -67,7 +67,12 @@ export class CoreWebService {
 
         return this._http.request(request).pipe(
             map((resp: Response) => {
-                return hasContent(resp) ? resp.json() : resp;
+                // some endpoints have empty body.
```
Comment (alfredo-dotcms) [inline projects/dotcms-js/src/lib/core/core-web.service.ts]: remove hasContent from `import`

**Pull request merged**